pkg/secret_provider/unmanaged_secret_provider.go

```diff
@@ -42,14 +49,37 @@ func newUnmanagedSecretProvider(logger *zap.Logger) (*UnmanagedSecretProvider, e
 }
 
 // initUnmanagedSecretProvider ...
-func initUnmanagedSecretProvider(logger *zap.Logger, kc *k8s_utils.KubernetesClient) (*UnmanagedSecretProvider, error) {
+func initUnmanagedSecretProvider(logger *zap.Logger, kc k8s_utils.KubernetesClient) (*UnmanagedSecretProvider, error) {
 	authenticator, authType, err := auth.NewAuthenticator(logger, kc)
 	if err != nil {
 		logger.Error("Error initializing unmanaged secret provider", zap.Error(err))
 		return nil, err
 	}
+
+	tokenExchangeURL, err := config.FrameTokenExchangeURL(kc, logger)
+	if err != nil {
+		logger.Error("Error fetching token exchange URL", zap.Error(err))
+		return nil, err
+	}
+	authenticator.SetURL(tokenExchangeURL)
+
+	if authenticator.IsSecretEncrypted() {
+		logger.Error("Secret is encrypted, decryption is only supported by sidecar container")
+		return nil, utils.Error{Description: localutils.ErrDecryptionNotSupported}
+	}
+
+	// Checking if the secret(api key) needs to be decoded
+	if authType == utils.DEFAULT && os.Getenv("IS_SATELLITE") == "True" {
+		logger.Info("Decoding apiKey since it's a satellite cluster")
+		decodedSecret, err := base64.StdEncoding.DecodeString(authenticator.GetSecret())
+		if err != nil {
+			logger.Error("Error decoding the secret", zap.Error(err))
+			return nil, err
+		}
+		authenticator.SetSecret(strings.TrimSuffix(string(decodedSecret), "\n"))
```

Comment:
can you add the comments why this trim is required

Reply:
Added comments.